pkg/asset/ignition/spec2to3.go

```diff
@@ -0,0 +1,74 @@
+package ignition
+
+import (
+	"encoding/json"
+
+	"github.com/pkg/errors"
+)
+
+// ConvertSpec2ToSpec3 converts Ignition spec2 to ignition spec3
+func ConvertSpec2ToSpec3(spec2data []byte) ([]byte, error) {
+	// Unmarshal
+	jsonMap := make(map[string]interface{})
+	err := json.Unmarshal(spec2data, &jsonMap)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to Marshal Ignition config")
+	}
+
+	// Replace ignition.version
+	ign := jsonMap["ignition"].(map[string]interface{})
+	ign["version"] = "3.0.0"
+
+	// ignition.config.append -> ignition.config.merge
+	config := ign["config"].(map[string]interface{})
+	if val, ok := config["append"]; ok {
+		config["merge"] = val
+		delete(config, "append")
+	}
+	ign["config"] = config
+	jsonMap["ignition"] = ign
+
+	// Delete networkd section
+	if _, ok := jsonMap["networkd"]; ok {
+		delete(jsonMap, "networkd")
+	}
+
+	// Modify storage.files
+	if sval, ok := jsonMap["storage"]; ok {
+		storage := sval.(map[string]interface{})
+
+		if fval, ok := storage["files"]; ok {
+			files := fval.([]interface{})
+
+			updatedFiles := make([]interface{}, 0)
+
+			for i := range files {
+				file := files[i].(map[string]interface{})
+				// Remove filesystem
+				if _, ok := file["filesystem"]; ok {
+					delete(file, "filesystem")
+				}
+				// append is no longer a flag
+				if val, ok := file["append"]; ok {
+					if val == "true" {
```

Comment:
This needs to be val == true.

Context, had a try to run OKD4 on FCOS today. Had to fix this, but next bump will be "failed to unmarshal bootstrap.ign: json: cannot unmarshal object into Go struct field File.append of type bool" as it tries to reread the config at some point and uses the old v2_2 config as template. There doesn't seem to be easy fix for that.

Reply:
Right, the installer would still use v2 ignition when creating new S3 buckets via terraform.

It doesn't look like installer can support both spec2 and spec3